### ISSUE TITLE
StreamChannel Media Archive not created when not "isValid"

### ISSUE THREAD

**Issue by njam** [cargomedia/cm#2038]
Noticed this today that in a `Session::deactivate()` context we don't create archives, because the publish has:
```php
    protected function _onDeleteBefore() {
        $streamChannel = $this->getStreamChannel();
        if ($streamChannel->isValid()) {
            $streamChannel->onUnpublish($this);
        }
    }
```

@alexispeter how about we move archiving to a separate function, independent of `onUnpublish()`?

**Comment by alexispeter** [comment]
How about we remove the isValid check from onDeleteBefore()? Maybe also remove it from StreamSubscribe. I looked at implementations of isValid() and onUnpublish() and don't think it would break anything. 
Furthermore, isValid() is something that should only be considered when it comes to subscribing/publishing to a streamChannel, but not unsubscribe/unpublishing imo.

**Comment by njam** [comment]
`VideoStreamChannel::onUnpublish()` would fail on deleted user?

**Comment by alexispeter** [comment]
What does that have to do with it? VideoStreamChannel::onUnpublish() will always fail when user is deleted, no matter if streamCHannel itself is valid or not.

**Comment by njam** [comment]
True, it's not covered by `isValid()`.
Ok, let's remove the check!

But probably `onPublish()` check check if `$streamPublish->getUser()` is `null`? Could you adjust it?

**Comment by alexispeter** [comment]
This should be fixed by #2039 as we will trigger archive-creation on delete rather than on unpublish.
